### PR TITLE
nixosTest.gocd-{agent,server}: port to python

### PR DESCRIPTION
**Pull request by flokli**
###### Motivation for this change
#72828

These already failed on master, but according to https://github.com/NixOS/nixpkgs/pull/61309, the service seems to behave fine.

The tests might just need a small update.

Looking at the logs it seems like the agent tries to download `agent-launcher.jar` from `https://localhost:8154`, is the https endpoint and not enabled. This might be just a matter of reconfiguring it in there, not sure about the error message in the PR here.

Could you take a look at it? This might also help in unbreaking https://github.com/NixOS/nixpkgs/pull/61309.

cc @cryptix 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by worldofpeace** [comment]
@GrahamcOfBorg test gocd-agent gocd-server

**Review by tfc — APPROVED** [review]

**Comment by flokli** [comment]
As written in https://github.com/NixOS/nixpkgs/pull/61309#issuecomment-557932078, these currently fail on master, and obviously still do after the refactor.

Would be nice if the maintainers @grahamc and @swarren83 could take a look into this.

**Comment by flokli** [comment]
I took a look. Turns out, the slightly weird hostnames `gocd_server` and `gocd_agent` were changed to `gocdserver` and `gocdagent` inside java, which failed name resolution, causing the tests to fail.

@GrahamcOfBorg test gocd-agent gocd-server

**Comment by flokli** [comment]
The `gocd-agent` test still fails, but the `gocd-server` test succeeds.

Turns out, now it's mandatory to pass some credentials while registering agents to a server.

The gocd server adds an `autoregister` key to the config file it generates, which would need to be passed to the client, after the server has spun up.

As I don't maintain the gocd NixOS modules, and only intended to port these tests to python (which were entirely broken before) I don't feel like I should pick this up myself, and leave it up to the maintainers.

I marked the gocd-agent test as broken, because that's what it still is.

**Pull request merged**
